From 9e504c1ba43f6492f96cbaf03b8dfadcc4152d41 Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Fri, 25 Oct 2024 17:05:15 +0800
Subject: [PATCH 243/265] Revert "mtk: mt76: mt7990: disable PPDU TxS to
 prevent MCU command timeout"

This reverts commit 6e8b3a9facc2f72c3fc7121bf3e0d1956ec1a2a2.

The root cause that TxS lead to MCU command timeout is found.
TxS is sent to host by TxFree ring so driver need to allocate rings and
set correct prefetch size. If hw cannot send TxS to host, it may lead
to TxS congestion on the hardware and hardware hang.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt7996/init.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/mt7996/init.c b/mt7996/init.c
index 687879be..8f0a2751 100644
--- a/mt7996/init.c
+++ b/mt7996/init.c
@@ -588,9 +588,7 @@ mt7996_mac_init_band(struct mt7996_dev *dev, u8 band)
 	/* MT_TXD5_TX_STATUS_HOST (MPDU format) has higher priority than
 	 * MT_AGG_ACR_PPDU_TXS2H (PPDU format) even though ACR bit is set.
 	 */
-	/* FIXME: In mt7990, it would lead to command timeout */
-	if (!is_mt7990(&dev->mt76))
-		mt76_set(dev, MT_AGG_ACR4(band), MT_AGG_ACR_PPDU_TXS2H);
+	mt76_set(dev, MT_AGG_ACR4(band), MT_AGG_ACR_PPDU_TXS2H);
 
 	if (!is_mt7996(&dev->mt76))
 		mt7996_mcu_set_bssid_mapping_addr(&dev->mt76, band);
-- 
2.45.2

